### PR TITLE
Repair clean build

### PR DESCRIPTION
**Pull request by jamesonwilliams**
`./gradlew clean build` wasn't completing successfully, [here's an example build scan](https://scans.gradle.com/s/bmeesvwv6djyk). The issue seems related to this Gradle issue with ANTLR: https://github.com/gradle/gradle/issues/19555

Adding a task dependency seems to resolve it. [Here's a build scan showing the build after this commit](https://gradle.com/s/tgy2ddpjfjgxe).

Resolves: https://github.com/square/gradle-dependencies-sorter/issues/38

### CODE REVIEW

File: grammar/build.gradle [hunk 1]
```diff
@@ -19,6 +19,7 @@ tasks.named('generateGrammarSource', AntlrTask) {
       // Specify the location of "libs"; i.e., for grammars composed of multiple files
       '-lib', antlrSrc
   ]
+  tasks.findByPath(":grammar:sourcesJar").dependsOn(it)
```
Comment (ZacSweers) [inline grammar/build.gradle:22]: This eagerly configures the task during the `generateGrammarSource` task's configuration. A better solution would be to do something like this outside this named block, which leverages lazy task configuration.

```kotlin
tasks.named(":grammar:sourcesJar") {
  dependsOn("generateGrammarSource")
}
```

**Review by ZacSweers — CHANGES_REQUESTED** [review]

File: grammar/build.gradle (hunk 1 above)
Comment (jamesonwilliams) [reply grammar/build.gradle:22]: It seems to work when I pull it out of the `generateGrammarSource` block:
```diff
   ]
-  tasks.findByPath(":grammar:sourcesJar").dependsOn(it)
 }
+tasks.findByPath(":grammar:sourcesJar").dependsOn("generateGrammarSource")
```
However, using the `named` block isn't working:
```diff
   ]
-  tasks.findByPath(":grammar:sourcesJar").dependsOn(it)
+}
+tasks.named(":grammar:sourcesJar") {
+  dependsOn("generateGrammarSource")
 }
```
```console
$ ./gradlew clean build

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/jameson.williams/gradle-dependencies-sorter/grammar/build.gradle' line: 23

* What went wrong:
A problem occurred evaluating project ':grammar'.
> Task with name ':grammar:sourcesJar' not found in project ':grammar'.
[...snip...]
```
I can update to the first version? WDYT?

Comment (ZacSweers) [reply grammar/build.gradle]: First version has the same problem (findByPath is an eager API). Second one is the right solution sorry I should've just put "sourcesJar" and not the gradle project path with it. That should work?

Comment (jamesonwilliams) [reply grammar/build.gradle]: It works! https://github.com/square/gradle-dependencies-sorter/pull/39/commits/f88909b1d1ec4ecfd12ee7a171af4b72e37efa20

File: grammar/build.gradle [hunk 2]
```diff
@@ -20,6 +20,9 @@ tasks.named('generateGrammarSource', AntlrTask) {
       '-lib', antlrSrc
   ]
 }
+tasks.named("sourcesJar") {
```
Comment (ZacSweers) [inline grammar/build.gradle:23]: Let's link the antlr issue for posterity

Comment (jamesonwilliams) [reply grammar/build.gradle:24]: Done in https://github.com/square/gradle-dependencies-sorter/pull/39/commits/ded65e9b377615fea8f7da6b9e932d39d30885c8

**Review by ZacSweers — APPROVED** [review]

**Pull request merged**